### PR TITLE
[dotnet] Add dotnet-linker, a project to contain custom linker steps for our net5.0 build.

### PR DESCRIPTION
**Pull request by rolfbjarne**
This PR is best reviewed commit-by-commit.

### CODE REVIEW

**Comment by spouliot** [comment]
@chamons it's not the `SetupStep` code from the net5 branch and, TBH, I'm not sure this step will (or should) survive - but it made things quite easier at this stage.

**Comment by rolfbjarne** [comment]
> I assume "Hello SetupStep" will be changed later?

Yes, that will be removed later when actual steps are implemented.

File: tools/dotnet-linker/SetupStep.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+using Mono.Linker.Steps;
+
+namespace Xamarin {
+
+	public class SetupStep : BaseStep {
+
+		protected override void Process ()
+		{
+			Console.WriteLine ("Hello SetupStep");
```
Comment (spouliot) [inline tools/dotnet-linker/SetupStep.cs:11]: maybe add a comment this is temporary ?
or a README.md in the directory ?
normally `master` is releasable - and that's not the case for this code yet

**Comment by rolfbjarne** [comment]
Test failure is unrelated.

**Pull request merged**